sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala

```diff
@@ -61,7 +61,8 @@ private[hive] object IsolatedClientLoader extends Logging {
     val files = if (resolvedVersions.contains((resolvedVersion, hadoopVersion))) {
       resolvedVersions((resolvedVersion, hadoopVersion))
     } else {
-      val remoteRepos = sparkConf.get(SQLConf.ADDITIONAL_REMOTE_REPOSITORIES)
+      val remoteRepos = sys.env.getOrElse(
+        "DEFAULT_ARTIFACT_REPOSITORY", sparkConf.get(SQLConf.ADDITIONAL_REMOTE_REPOSITORIES))
```

Comment:
This still isn't quite right -- `remoteRepos` still represents the repositories to be tried _after_ the default Maven Central repository. I looked more closely at this and actually the changes you've made in `SparkSubmit` should already cover this scenario. `IsolatedClientLoader#downloadVersion()` calls `SparkSubmitUtils.resolveMavenCoordinates()` using `SparkSubmitUtils.buildIvySettings()`, which in turn calls `SparkSubmitUtils.createRepoResolvers()` to get the default repository, which is where you've already made your changes.

Reply:
@xkrogen During my testing of your suggested changes, the test still tries to download the artifact from `SQLConf.ADDITIONAL_REMOTE_REPOSITORIES` which points to `https://maven-central.storage-download.googleapis.com/maven2/`. I still need the change in `SQLConf.scala` to overwrite the maven repo. 

```   val ADDITIONAL_REMOTE_REPOSITORIES =
    buildConf("spark.sql.maven.additionalRemoteRepositories")
      .doc("A comma-delimited string config of the optional additional remote Maven mirror " +
        "repositories. This is only used for downloading Hive jars in IsolatedClientLoader " +
        "if the default Maven Central repo is unreachable.")
      .version("3.0.0")
      .stringConf
      .createWithDefault(
        sys.env.getOrElse(
          "DEFAULT_ARTIFACT_REPOSITORY",
          "https://maven-central.storage-download.googleapis.com/maven2/"))```

Let me know your thoughts on this.

Reply:
IMO if you want to fully change the repository, you should be configuring both `DEFAULT_ARTIFACT_REPOSITORY` and `spark.sql.maven.additionalRemoteRepositories`. I think having a config whose default value changes based on an environment variable is confusing behavior.

Reply:
I was looking at something similar today and realized that what I said doesn't quite make sense, since you can't really externally configure `spark.sql.maven.additionalRemoteRepositories` when running the tests via Maven.

I'm still not sure if configuring the default value of a config based on an environment variable makes sense, though. Maybe it would be better to update the tests somewhere (if that is indeed your concern).

Reply:
`MAVEN_OPTS` should do the trick, no ?

Reply:
Yeah it seems so @mridulm -- after I wrote my last comment I started to wonder if that was correct, then saw in the documentation that Spark will pick up config values from System properties. Still trying to make it work in my environment, but I think you are right.

Reply:
@mridulm So it is a bit more complicated than this. We need the options on the Surefire JVM, not the Maven JVM, so it's actually proper to use a flag like `-DargLine="-Dspark.sql.maven.additionalRemoteRepositories=..."`. In my tests this had some negative interaction with the Hive module, which overrides the default `argLine` with a custom one to disable assertions.

Even if that problem can be solved, the Hive test suite launches _entirely separate processes_ via `spark-submit`, so even `argLine` is insufficient to pass along configs.

I was able to get it picked up by setting `JAVA_TOOL_OPTIONS="-Dspark...."`, but this affects every single JVM and is a bit messy.